spec/lib/reports/suppliers/pay_your_suppliers_report_spec.rb

```diff
@@ -49,23 +49,51 @@
 
   context "with taxes and fees" do
     let(:line_item) { order.line_items.first }
-    let!(:fees_adjustment) do
-      create(:adjustment, originator_type: "EnterpriseFee", adjustable: line_item, amount: 0.1)
+    let!(:enterprise_fee) do
+      create(
+        :enterprise_fee,
+        enterprise: supplier,
+        fee_type: 'sales',
+        amount: 0.1,
+      )
     end
     let!(:tax_adjustment) do
       create(:adjustment, originator_type: "Spree::TaxRate", adjustable: line_item, amount: 0.1)
     end
 
+    before do
+      hub.update!(charges_sales_tax: true)
+      supplier.update!(charges_sales_tax: true)
+
+      exchange = order_cycle.exchanges.take
+      exchange.enterprise_fees << enterprise_fee
+      exchange.exchange_variants.build(variant: line_item.variant)
+      exchange.incoming = true
+      exchange.save!
+
+      OpenFoodNetwork::EnterpriseFeeCalculator
+        .new(hub, order_cycle)
+        .create_line_item_adjustments_for(line_item)
+
+      fees_adjustment = line_item.adjustments.enterprise_fee.first
+      create(
+        :adjustment,
+        originator_type: "Spree::TaxRate",
+        adjustable: fees_adjustment,
+        amount: 0.01
+      )
+    end
```

Comment:
It feels a bit risky to be creating the fee adjustment directly here. What if we've done it the wrong way, or in the future, the app creates fees in a different way?

It looks like the system should be doing that for us (you already created the 'sales' fee above), but I guess it didn't work. `±` So I think we should find a way to make it work.

Reply:
Sure, that sounds good. It's incorporated here such that every adjustment is created by the order itself:
https://github.com/openfoodfoundation/openfoodnetwork/pull/12879/commits/b09a79571829b05ff963ef60a4ba47f62edce089